unity-renderer/Assets/DCLServices/WearablesCatalogService/LambdasWearablesCatalogService.cs

```diff
@@ -397,64 +398,90 @@ private async UniTask<WearableItem> SyncWearablesRequestsAsync(string newWearabl
             return result.FirstOrDefault(x => x.id == newWearableId);
         }
 
-        private static IList<WearableDefinition> ValidateWearables(IList<WearableDefinition> wearablesFromLambdas)
+        private List<WearableItem> ValidateWearables(IEnumerable<WearableWithEntityResponseDto.ElementDto> wearableElements)
         {
-            var invalidWearablesIndices = ListPool<int>.Get();
+            List<WearableItem> wearables = new ();
 
-            for (var i = 0; i < wearablesFromLambdas.Count; i++)
+            foreach (var item in wearableElements)
             {
-                var item = wearablesFromLambdas[i];
-                var definition = item.definition;
+                WearableWithEntityResponseDto.ElementDto.EntityDto entity = item.entity;
+                WearableWithEntityResponseDto.ElementDto.EntityDto.MetadataDto metadata = entity.metadata;
 
-                if (FilterInvalidWearableItem(item, i, invalidWearablesIndices))
+                if (IsInvalidWearable(metadata))
                     continue;
 
                 try
                 {
-                    foreach (var representation in definition.data.representations)
+                    WearableItem wearable = new WearableItem
```

Comment:
Couldnt we deprecate this old WearableItem model and use directly the MetadataDto one?